optimum/onnxruntime/modeling_ort.py

```diff
@@ -1648,6 +1662,352 @@ def _prepare_onnx_inputs(self, use_torch: bool, **kwargs):
+                "input_values": input_values.cpu().detach().numpy(),
+            }
+
+            # run inference
+            outputs = self.model.run(None, onnx_inputs)
+            logits = torch.from_numpy(outputs[self.output_names["logits"]]).to(self.device)
+            embeddings = torch.from_numpy(outputs[self.output_names["embeddings"]]).to(self.device)
+
+            # converts output to namedtuple for pipelines post-processing
+            return XVectorOutput(logits=logits, embeddings=embeddings)
+
+
+AUDIO_FRAME_CLASSIFICATION_EXAMPLE = r"""
+    Example of audio frame classification:
+
+    ```python
+    >>> from transformers import {processor_class}
+    >>> from optimum.onnxruntime import {model_class}
+    >>> from datasets import load_dataset
+    >>> import torch
+
+    >>> dataset = load_dataset("hf-internal-testing/librispeech_asr_demo", "clean", split="validation")
+    >>> dataset = dataset.sort("id")
+    >>> sampling_rate = dataset.features["audio"].sampling_rate
+
+    >>> feature_extractor = {processor_class}.from_pretrained("{checkpoint}")
+    >>> model =  {model_class}.from_pretrained("{checkpoint}")
+
+    >>> inputs = feature_extractor(dataset[0]["audio"]["array"], return_tensors="pt", sampling_rate=sampling_rate)
+    >>> with torch.no_grad():
+    ...     logits = model(**inputs).logits
+
+    >>> probabilities = torch.sigmoid(logits[0])
+    >>> labels = (probabilities > 0.5).long()
+    >>> labels[0].tolist()
+    ```
+"""
+
+
+@add_start_docstrings(
+    """
+    Onnx Model for with a frame classification head on top for tasks like Speaker Diarization.
+    """,
+    ONNX_MODEL_START_DOCSTRING,
+)
+class ORTModelForAudioFrameClassification(ORTModel):
+    """
+    Audio Frame Classification model for ONNX.
+    """
+
+    auto_model_class = AutoModelForAudioFrameClassification
+
+    @add_start_docstrings_to_model_forward(
+        ONNX_AUDIO_INPUTS_DOCSTRING.format("batch_size, sequence_length")
+        + AUDIO_FRAME_CLASSIFICATION_EXAMPLE.format(
+            processor_class=_FEATURE_EXTRACTOR_FOR_DOC,
+            model_class="ORTModelForAudioFrameClassification",
+            checkpoint="optimum/wav2vec2-base-superb-sd",
+        )
+    )
+    def forward(
+        self,
+        input_values: Optional[torch.Tensor] = None,
+        **kwargs,
+    ):
+        if self.device.type == "cuda" and self.use_io_binding:
+            io_binding, output_shapes, output_buffers = self.prepare_io_binding(
+                input_values, ordered_input_names=self._ordered_input_names
+            )
+
+            # run inference with binding & synchronize in case of multiple CUDA streams
+            io_binding.synchronize_inputs()
+            self.model.run_with_iobinding(io_binding)
+            io_binding.synchronize_outputs()
+
+            # converts output to namedtuple for pipelines post-processing
+            return TokenClassifierOutput(logits=output_buffers["logits"].view(output_shapes["logits"]))
```

Comment:
So if I read correctly, this is not tested in test_modeling.py. Could we rather raise a `NotImplementedError` here?

Reply:
Sure